components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/JITProvisioningPostAuthenticationHandler.java

```diff
@@ -426,12 +433,49 @@ private PostAuthnHandlerFlowStatus handleRequestFlow(HttpServletRequest request,
                     }
                     callDefaultProvisioningHandler(username, context, externalIdPConfig, localClaimValues,
                             stepConfig);
+
+                    // Trigger event to store last login time after successful authentication.
+                    if (StringUtils.isEmpty(associatedLocalUser)) {
+                        triggerEvent(username, context, username);
+                    } else {
+                        triggerEvent(associatedLocalUser, context, username);
+                    }
                 }
             }
         }
         return SUCCESS_COMPLETED;
     }
 
+    private void triggerEvent(String username, AuthenticationContext context, String emailUsername)
+            throws PostAuthenticationFailedException {
+
+        if (StringUtils.isEmpty(username)) {
+            return;
+        }
+
+        try {
+            UserRealm realm = getUserRealm(context.getTenantDomain());
+            UserStoreManager userStoreManager = getUserStoreManager(context.getExternalIdP()
+                    .getProvisioningUserStoreId(), realm, emailUsername);
+            IdentityEventService eventService = FrameworkServiceDataHolder.getInstance().getIdentityEventService();
+            Map<String, Object> eventProperties = new HashMap<>();
+            eventProperties.put(IdentityEventConstants.EventProperty.USER_NAME, username);
+            eventProperties.put(IdentityEventConstants.EventProperty.USER_STORE_MANAGER, userStoreManager);
+            eventProperties.put(IdentityEventConstants.EventProperty.OPERATION_STATUS, true);
+            Event event = new Event(IdentityEventConstants.Event.JIT_USER_SET_CLAIMS, eventProperties);
```

Comment:
change to two events one for Jit user add, and one for Jit user update. and fire those events separately. In the event handler, you may treat the two events in the same manner for this feature

Reply:
Decided to go with the current implementation